libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3271,8 +3283,12 @@ void TypeChecker::requireLValue(Expression const& _expression, bool _ordinaryAss
 				if (varDecl->isExternalCallableParameter() && dynamic_cast<ReferenceType const*>(identifier->annotation().type))
 					return { 7128_error, "External function arguments of reference type are read-only." };
 
+		fatal = true;
 		return { 4247_error, "Expression has to be an lvalue." };
 	}();
 
-	m_errorReporter.typeError(errorId, _expression.location(), description);
+	if (fatal)
+		m_errorReporter.fatalTypeError(errorId, _expression.location(), description);
```

Comment:
This was done so that the following test case was creating problems https://github.com/ethereum/solidity/pull/9146/files#diff-34333568973cd82c6d17c24bcc4c1ea8 The inner `RationalNumberType` was typechecked and the assert `nameable()` failed.

Reply:
Can you show the full call trace? Do you think we can still "fix" this?

Reply:
Fixed!